cmd/login/auth.go

```diff
@@ -203,6 +187,33 @@ func printMCCredentials(k8sConfigAccess clientcmd.ConfigAccess, i *installation.
 	return nil
 }
 
+// oidcExec creates an ExecConfig for OIDC authentication using the credential plugin framework
+func oidcExec(issuerURL, clientID, refreshToken string) *clientcmdapi.ExecConfig {
+	// Get the path to the kubectl-gs binary
+	// We use "kubectl-gs" as the command
+	execPath := "kubectl-gs"
+
+	return &clientcmdapi.ExecConfig{
+		APIVersion: "client.authentication.k8s.io/v1beta1",
+		Command:    execPath,
+		Args:       []string{"credential-plugin"},
+		Env: []clientcmdapi.ExecEnvVar{
+			{
+				Name:  "KUBECTL_GS_OIDC_ISSUER_URL",
+				Value: issuerURL,
+			},
+			{
+				Name:  "KUBECTL_GS_OIDC_CLIENT_ID",
+				Value: clientID,
+			},
+			{
+				Name:  "KUBECTL_GS_OIDC_REFRESH_TOKEN",
+				Value: refreshToken,
```

Comment:
These are [declared consts](https://github.com/giantswarm/kubectl-gs/pull/1798/changes#diff-8a70f7bff92dcf65f8d2b37d2bee8d0ab31b3fdcdde72cd4a8b34a5f88b37547R23) in `credentialplugin` so you could re-use or extract them to a key file, but tbh I don't mind if you just keep them as strings here.